jetty-core/jetty-websocket/jetty-websocket-jetty-common/src/main/java/org/eclipse/jetty/websocket/common/JettyWebSocketFrameHandler.java

```diff
@@ -315,6 +324,14 @@ private void onPingFrame(Frame frame, Callback callback)
         }
         else
         {
+            // If we have a frameHandler it takes responsibility for handling the ping and demanding.
+            if (frameHandle != null)
+            {
+                callback.succeeded();
+                autoDemand();
+                return;
+            }
```

Comment:
This is now forcing applications to send a PONG from `onWebSocketFrame()` - I don't think it's nice.

Reply:
If I automatically send the pong after this, then its not possible to do everything from the `onFrame`.

For example you could not implement a WS proxy just from `onFrame` you would need both `onFrame` and `onPong`.